pkg/clusterCache/ClusterCache.go

```diff
@@ -0,0 +1,256 @@
+	var cache clustercache.ClusterCache
+	var ok bool
+	impl.rwMutex.RLock()
+	cache, ok = impl.clustersCache[clusterInfo.ClusterId]
+	impl.rwMutex.RUnlock()
+
+	if ok {
+		return cache, nil
+	}
+	clusterConfig := clusterInfo.GetClusterConfig()
+	restConfig, err := impl.k8sUtil.GetRestConfigByCluster(clusterConfig)
+	if err != nil {
+		impl.logger.Errorw("error in getting rest config", "err", err, "clusterName", clusterConfig.ClusterName)
+		return cache, err
+	}
+	cache = clustercache.NewClusterCache(restConfig, getClusterCacheOptions()...)
+	return cache, nil
+}
+
+func getClusterCacheOptions() []clustercache.UpdateSettingsFunc {
+	clusterCacheOpts := []clustercache.UpdateSettingsFunc{
+		clustercache.SetListSemaphore(semaphore.NewWeighted(clusterCacheListSemaphoreSize)),
+		clustercache.SetListPageSize(clusterCacheListPageSize),
+		clustercache.SetListPageBufferSize(clusterCacheListPageBufferSize),
+		clustercache.SetWatchResyncTimeout(clusterCacheWatchResyncDuration),
+		clustercache.SetClusterSyncRetryTimeout(clusterSyncRetryTimeoutDuration),
+		clustercache.SetResyncTimeout(clusterCacheResyncDuration),
+		clustercache.SetPopulateResourceInfoHandler(func(un *unstructured.Unstructured, isRoot bool) (interface{}, bool) {
+			fmt.Println("resource updated")
+			gvk := un.GroupVersionKind()
+			res := &bean.ResourceNode{
+				Port:            getPorts(un, gvk),
+				ResourceVersion: un.GetResourceVersion(),
+				NetworkingInfo: &bean.ResourceNetworkingInfo{
+					Labels: un.GetLabels(),
+				},
+			}
+
+			if k8sUtils.IsService(gvk) && un.GetName() == k8sUtils.DEVTRON_SERVICE_NAME && k8sUtils.IsDevtronApp(res.NetworkingInfo.Labels) {
+
+				res.Health = &bean.HealthStatus{
+					Status: bean.HealthStatusHealthy,
+				}
+			} else {
+				if healthCheck := health.GetHealthCheckFunc(gvk); healthCheck != nil {
+					health, err := healthCheck(un)
+					if err != nil {
+						res.Health = &bean.HealthStatus{
+							Status:  bean.HealthStatusUnknown,
+							Message: err.Error(),
+						}
+					} else if health != nil {
+						res.Health = &bean.HealthStatus{
+							Status:  string(health.Status),
+							Message: health.Message,
+						}
+					}
+				}
+			}
+			if k8sUtils.IsPod(gvk) {
+				infoItems, _ := argo.PopulatePodInfo(un)
+				res.Info = infoItems
+			}
+			// hibernate set starts
+			if un.GetOwnerReferences() == nil {
+				// set CanBeHibernated
+				replicas, found, _ := unstructured.NestedInt64(un.UnstructuredContent(), "spec", "replicas")
+				if found {
+					res.CanBeHibernated = true
+				}
+
+				// set IsHibernated
+				annotations := un.GetAnnotations()
+				if annotations != nil {
+					if val, ok := annotations[hibernateReplicaAnnotation]; ok {
+						if val != "0" && replicas == 0 {
+							res.IsHibernated = true
+						}
+					}
+				}
+			}
+			// hibernate set ends
+
+			return res, gvk.Kind == kube.CustomResourceDefinitionKind
```

Comment:
why do we want to set whole manifest for CRDs ?? ref: `gvk.Kind == kube.CustomResourceDefinitionKind`